adapters/repos/db/index_queue.go

```diff
@@ -339,43 +339,35 @@ func (q *IndexQueue) Size() int64 {
 	return count
 }
 
-// Delete marks the given vectors as deleted.
-// This method can be called even if the async indexing is disabled.
+// Deletes the vectors from the index synchronously
+// if they are already indexed, otherwise it marks them as deleted in the queue.
+// If async indexing is disabled, it calls the index delete method directly.
 func (q *IndexQueue) Delete(ids ...uint64) error {
-	if !asyncEnabled() {
-		return q.index.Delete(ids...)
+	if len(ids) == 0 {
+		return nil
 	}
+
 	start := time.Now()
 	defer q.metrics.Delete(start, len(ids))
 
+	if !asyncEnabled() {
+		return q.index.Delete(ids...)
+	}
+
 	q.indexLock.RLock()
 	defer q.indexLock.RUnlock()
 
-	remaining := make([]uint64, 0, len(ids))
-	indexed := make([]uint64, 0, len(ids))
-
-	for _, id := range ids {
-		if q.index.ContainsNode(id) {
-			indexed = append(indexed, id)
-
-			// is it already marked as deleted in the queue?
-			if q.queue.IsDeleted(id) {
-				q.queue.ResetDeleted(id)
+	for i := range ids {
+		if q.index.ContainsNode(ids[i]) {
+			err := q.index.Delete(ids[i])
```

Comment:
**question:**  given that we don't act on error 
can't we pass `q.index.Delete(ids)` all at once ?

Reply:
This way is slightly more performant because it ensure we don't have a tombstone already, whereas calling index.Delete only checks if the id exists, which may incur unnecessary commit log lines